### PR TITLE
Add more examples on the Rust API

### PR DESCRIPTION
**Pull request by jubianchi**
# Description

* [x] Clean and comment the memory example (#174)
* [x] Move examples from `wasmerio/docs.wasmer.io` here:
  * [x] Early exit (comments to add)
  * [x] Error handling
  * [x] Passing data (env)

# Review

- [ ] Add a short description of the the change to the CHANGELOG.md file

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -37,245 +37,288 @@ The examples are written in a difficulty/discovery order. Concepts that
...
    </details>
 
+### Tunables
 
-10. [**Exported memory**][exported-memory], explains how to read from 
-    and write to exported memory.
-   
-   _Keywords_: export, memory.
+1. **Limit memory**, explains how to use Tunables to limit the
+   size of an exported Wasm Memories
```
Comment (webmaster128) [inline examples/README.md:194]:
```suggestion
   size of an exported Wasm memory.
```

File: examples/errors.rs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    println!("Calling `div_by_zero` function...");
+    // Let's call the `div_by_zero` exported function.
+    let result = div_by_zero.call();
+
+    // When we call a function it can either succeed or fail. We expect it to fail.
+    match result {
+        Ok(_) => {
+            // This should have thrown an error, return an error
+            panic!("throw_wasm_error did not error");
```
Comment (webmaster128) [inline examples/errors.rs:75]:
```suggestion
            panic!("div_by_zero did not error");
```

File: examples/errors.rs [hunk 3]
```diff
@@ -0,0 +1,103 @@
+//! A Wasm module can sometimes not work, it will produce errors in some
+//! cases.
```
Comment (syrusakbary) [inline examples/errors.rs:2]: //! A Wasm module can sometimes be invalid or trigger traps, and in those case we will get
//! an error back from the API.

File: examples/instance.rs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    // Create an empty import object.
+    let import_object = imports! {};
+
+    println!("Instantiating module...");
+    // Let's instantiate the Wasm module.
+    let instance = Instance::new(&module, &import_object)?;
+
+    // We now have an instance ready to be used.
+    //
+    // From an `Instance` we can fetch any exported entities from the Wasm module.
```
Comment (syrusakbary) [inline examples/instance.rs:57]:
```suggestion
    // From an `Instance` we can retrieve any exported entities.
```

File: examples/instance.rs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    // Each of these entities is covered in others examples.
+    //
+    // Here we are fetching the exported function. We won't go into details here
```
Comment (syrusakbary) [inline examples/instance.rs:60]:
```suggestion
    // Here we are retrieving the exported function. We won't go into details here
```

**Review by syrusakbary — APPROVED** [review]

**Review by syrusakbary — APPROVED** [review]

**Pull request merged**